docs/python/tutorial-azure-functions.md

```diff
@@ -0,0 +1,348 @@
+    logging.info('Python HTTP trigger function processed a request.')
+
+    name = req.params.get('name')
+    if not name:
+        try:
+            req_body = req.get_json()
+        except ValueError:
+            pass
+        else:
+            name = req_body.get('name')
+
+    if name:
+        return func.HttpResponse(f"Hello {name}!")
+    else:
+        return func.HttpResponse(
+             "Please pass a name on the query string or in the request body",
+             status_code=400
+        )
+```
+
+The important parts of the code are as follows:
+
+- You must import `func` from `azure.functions`; importing the logging module is optional but recommended.
+- The required `main` function receives a `func.request`  object named `req`, and returns a value of type `func.HttpResponse`. You can learn more about the capabilities of these objects in the [func.HttpRequest](https://docs.microsoft.com/python/api/azure-functions/azure.functions.httprequest?view=azure-python) and [func.HttpResponse](https://docs.microsoft.com/python/api/azure-functions/azure.functions.httpresponse?view=azure-python) references.
+- The body of `main` then does whatever it wants to process the request and generate a response. In this case, the function looks for a `name` parameter in the URL. Failing that, it checks if the request body contains JSON (using the `func.HttpRequest.get_json` function) and that the JSON contains a `name` value (using the `get` method of the JSON object returned by `get_json`).
+- If a name is found, the function returns the string "Hello" with the name appended; otherwise it returns an error message.
+
+## Test and debug the function locally
+
+1. When you create the Functions project, the VS Code extension also creates a launch configuration in `.vscode/launch.json` that contains a single configuration named **Attach to Python Functions**. This configuration means you can just press F5 or use the Debug explorer to start the project:
+
+    ![Debug explorer showing the Functions launch configuration](images/functions/launch-configuration.png)
+
+1. When you start the debugger, a terminal opens showing output from Azure Functions, including a summary of the available endpoints:
+
+    ```
+    Hosting environment: Production
+    Content root path: d:\Examples\Python\AzureFunctions
+    Now listening on: http://0.0.0.0:7071
+    Application started. Press Ctrl+C to shut down.
+
+    Http Functions:
+
+            HttpExample: [GET,POST] http://localhost:7071/api/HttpExample
+    ```
+
+1. Ctrl+click the URL in the output window to open a browser to that address, or start a browser and paste in the same URL. In either case, you can see that the endpoint is `api/<function_name>`, in this case `api/HttpExample`. However, because that URL doesn't include a name parameter, the browser window should just show, "Please pass a name on the query string or in the request body" as appropriate for that path in the code.
```

Comment:
Platform: Is this a different gesture on macOS? Not familiar enough with vscode on that platform to know.

Reply:
There's actually a keyboard encoding I need to use here. Thanks for catching it.